envconfig.go

```diff
@@ -198,7 +198,10 @@ func Process(prefix string, spec interface{}) error {
 		// here to use os.LookupEnv for >=go1.5
 		value, ok := lookupEnv(info.Key)
 		if !ok && info.Alt != "" {
-			value, ok = lookupEnv(info.Alt)
+			// if the key has not been set in the environment, use the `envconfig`
+			// tag value without the prefix
+			info.Key = info.Alt
```

Comment:
I dont follow why we are setting info.Key to info.Alt here.

Reply:
basically, `info.Key` contains either the key of the environment variable or the alternative value (which is `prefix`_ the value of the `envconfig` tag). however, there are situations where we want only the value of the `envconfig` tag without the `prefix` - this is `info.Alt`.

if we check for the `info.Key` in the environment variables and it's not there, we search for the `info.Alt` value. so I set the key to that value.

Reply:
Sorry, I'm still confused by this. Can you elaborate on the use cases where we need to omit the prefix? 

At this point in the code, if the `envconfig` tag is not empty (let's say the tag is `envconfig:"othername"`, then `info.Key` contains `prefix_othername` and `info.Alt` contains `othername`. The previous code was already doing the check that we are doing in this code block, but without overriding `info.Key`, why do we need to override it now?

Reply:
because at the parse error what we see as info.Key can be wrong. if the info.key is as you said `othername`, when a parse error occurs, what we have is `prefix_othername` and it didn't take into account that the value being used may not have been from `info.Alt`.

what my change does is to only override `info.Key` if we use the value from 1info.Alt` and carry that through

Reply:
a possible alternative could be to overwrite the `info.Key` only if the `info.Alt` variable is found

Reply:
ahh, gotcha! I completely missed the `info.Key` call in `ParseError`. Makes sense to me, thanks for clarifying!